src/main/groovy/io/gatling/gradle/GatlingEnterprisePackageTask.groovy

```diff
@@ -1,79 +1,88 @@
+                    resolvedDependency.module.id.version,
+                    moduleArtifact.file
+                ))
+        }.toSet()
     }
 
     @Classpath
-    FileCollection getIncludedDependencies() {
-        ResolvedConfiguration resolvedConfiguration = getResolvedConfiguration()
+    @Optional
+    List<Configuration> configurations
 
-        Set<ResolvedDependency> gatlingDependencies = new HashSet<ResolvedDependency>()
-        collectGatlingDepsRec(resolvedConfiguration.getFirstLevelModuleDependencies(), gatlingDependencies)
+    @Internal
+    BasePluginExtension basePluginExtension = project.extensions.getByType(BasePluginExtension)
 
-        project.files(resolvedConfiguration.files) - project.files(gatlingDependencies.collect {
-            it.moduleArtifacts*.file
-        }.flatten())
-    }
+    @OutputFile
+    final RegularFileProperty outputFile = project.objects.fileProperty().fileValue(getJarFile())
 
-    @Override
-    protected CopyAction createCopyAction() {
-        return new GatlingEnterpriseCopyAction(
-            this.archiveFile.get().asFile,
-            getMetadataCharset(),
-            getMainSpec().buildRootResolver().getPatternSet(),
-            isPreserveFileTimestamps(),
-            getEntryCompression(),
-            isZip64()
+    @TaskAction
+    void createArchive() {
+        EnterprisePackager packager = new EnterprisePackager(new GradlePluginIO(logger).getLogger())
+        ResolvedConfiguration resolvedConfiguration = getResolvedConfiguration()
+        packager.createEnterprisePackage(
+            getClassDirectories(),
+            collectGatlingDependencies(resolvedConfiguration.getFirstLevelModuleDependencies()),
+            collectExtraDependencies(resolvedConfiguration.getFirstLevelModuleDependencies()),
+            project.group,
+            project.name,
+            project.version,
+            'gradle',
+            getClass().getPackage().getImplementationVersion(),
+            outputFile.asFile.get()
         )
     }
 
-    private String gatlingVersion() {
-        for (artifact in getResolvedConfiguration().resolvedArtifacts.flatten()) {
-            def id = artifact.moduleVersion.id
-            if (id.group == "io.gatling" && id.name == "gatling-app") {
-                logger.debug("Detected Gatling compile version: {}", id.version)
-                return id.version
+
+    private File getJarFile() {
+        String jarVersion = skipVersionIfUnspecified((String) project.version)
+        new File(
+            basePluginExtension.libsDirectory.get().asFile.getAbsolutePath(),
+            project.name +
+                jarVersion +
+                '-' + ARCHIVE_CLASSIFIER +
+                JAR_EXTENSION)
+    }
+
+    private List<File> getClassDirectories() {
+        project.getAllprojects().collect { p ->
+            JavaPluginExtension javaPluginExtension = p.getExtensions().getByType(JavaPluginExtension.class)
```

Comment:
What if this project uses Scala or Kotlin?

Reply:
It's seems to be the plugin used by all the jvm related project.

As they themselve say in [the documentation](https://docs.gradle.org/current/javadoc/org/gradle/api/plugins/JavaPluginExtension.html): 
> This extension is added by the JavaBasePlugin and would be more appropriately named the JvmPluginExtension extension. It is used to configure many of the project's JVM-related settings and behavior.

I've just tested it with the gatling-gradle-plugin-demo-kotlin project.